### ISSUE TITLE
Use 100% Custom CSS for Gradio Components

### ISSUE THREAD

**Issue by yangyu** [gradio-app/gradio#8924]
- [ ] I have searched to see if a similar issue already exists.
Yes, it happened on both old and latest version of Gradio

**Is your feature request related to a problem? Please describe.**  
Yes, I find that all components will use the theme CSS somehow(custom CSS always been overwirte by others even with !important ) which limited the extention of use custom CSS. 

**Describe the solution you'd like**  
My thought is that If you can provide a flag to control whether use the theme CSS or not in components like :

button = gr.Button(ignore_theme_css=True,element_id="mybtn") 

If it is been set then do not use any CSS of default theme or any theme for this component. This will allow user to config their Custom CSS to complete fit to their style. 

**Additional context**

**Comment by abidlabs** [comment]
Hi @yangyu have you tried setting `theme="base"` in your Gradio app (it is a parameter in `Blocks`)?

**Comment by yangyu** [comment]
> Hi @yangyu have you tried setting `theme="base"` in your Gradio app (it is a parameter in `Blocks`)?

Yes, I tried. And I found that the css .dark will be apply anyway which will use default theme to overwrite all. So it is not possible to use custom css 100%.

**Comment by ketsapiwiq** [comment]
Hi, yes I have the same problem where I heavily rely on custom css and custom components to change the appearance.
I had also a weird behavior with the css .dark.
I would be so happy if there's just a way to disable ALL css defined by Gradio themes: a `Theme=None`.
So there would still be the CSS added by Svelte in each component I guess. But you can still fork them to create a no-style generic custom component I guess.
The problem is when there are added styles for top-level DOM elements that you have to counter with `!important`.

Thanks for any effort in that direction :)

**Comment by ketsapiwiq** [comment]
As a quick update: one of the main problems to me for custom theming is the duplicate CSS with the `gradio-app .gradio-container.gradio-container-4-38-1 .contain` class added, it's a pain to debug and I don't understand well the rationale / how it could be disabled.
Thank you :)

**Comment by ketsapiwiq** [comment]
Adding an example, I'm working on a custom component and my CSS is nicely scoped into Svelte, but there's this `.contain` injection that allows a `.contain button { color: inherit;}` to have more priority:

![Image](https://github.com/user-attachments/assets/dc70d1d9-ae29-429b-a5d2-07b99b54710f)

I realize only now this comes from importing my color: inherit css rule through `gr.Blocks(css=config.css)`.
That means I probably shouldn't put my reset styles here and instead maybe import it elsewhere with a `<style>` tag (or `<link>` ?) to have less priority.
But when I do that, then there's rules like the default Gradio `.prose a { text-decoration: underline; }` that take over when I least expect it (in a non-custom component), and I would need my reset rules to have a higher priority.

Thanks in advance, will stop bumping the thread I swear :p

**Comment by abidlabs** [comment]
I would suggest using `gr.HTML` components for this, which includes a `apply_default_css=False` argument: https://gradio.app/guides/custom-HTML-components
